### PR TITLE
Add type definitions for vinyl-named

### PR DESCRIPTION
**Pull request by MaikoTan**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/vinyl-named/vinyl-named-tests.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    // $ExpectType string
+    file.named;
+    return 'your own name';
+  }));
+
+fs.src('src/*.js')
+  .pipe(named(function (file) {
+    file.customName = 'your name'
+    this.queue(file)
+  }))
```
Comment (MaikoTan) [inline types/vinyl-named/vinyl-named-tests.ts:19]:
```suggestion
  .pipe(named(function(file) {
    file.customName = 'your name';
    this.queue(file);
  }));
```

File: types/vinyl-named/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Type definitions for vinyl-named 1.1
+// Project: https://github.com/shama/vinyl-named
+// Definitions by: Maiko Tan <https://github.com/MaikoTant>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
```
Comment (peterblazejewicz) [inline types/vinyl-named/index.d.ts:6]:
```suggestion
```
do you need this explicitely? The setup which uses this module already should have @types/node (say via gulp itself). Adding this explicitely impacts CI tests (ever increasing time).
thanks!

Comment (MaikoTan) [reply types/vinyl-named/index.d.ts]: I am not very familiar with this. Thanks for pointing that out! 🤣

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@MaikoTan thanks!

**Comment by MaikoTan** [comment]
@peterblazejewicz Thanks for the reviewing!
@typescript-bot Ready to merge

**Comment by MaikoTan** [comment]
Ready to merge

**Pull request merged**